src/thor/bidirectional_astar.cc

```diff
@@ -306,6 +306,7 @@ inline bool BidirectionalAStar::ExpandForwardInner(GraphReader& graphreader,
   edgelabels_forward_.emplace_back(pred_idx, meta.edge_id, opp_edge_id, meta.edge, newcost, sortcost,
                                    dist, mode_, transition_cost,
                                    (pred.not_thru_pruning() || !meta.edge->not_thru()),
+                                   (pred.closure_pruning() || !costing_->IsClosed(meta.edge, tile)),
```

Comment:
its unfortunate to have to check this twice. basically we already called IsClosed above potentially inside of the allowed function and now we call it again here. it will be interesting to see the performance impact of this in a system that is actually hooked up with live traffic (ie when calling this is non trivial)

Reply:
I have a static dump of traffic that I'm running perf tests with. Will post back the results here shortly.